sql/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -97,7 +97,7 @@ public LogicalPlan build(TableStats tableStats, Set<Symbol> usedBeforeNextFetch)
             JoinOperations.buildRelationsToJoinPairsMap(
                 JoinOperations.convertImplicitJoinConditionsToJoinPairs(mss.joinPairs(), queryParts));
 
-        final boolean hasOuterJoins = joinPairs.values().stream().anyMatch(p -> p.joinType().isOuter());
+        final boolean orderByCanBePushedDown = joinPairs.values().stream().noneMatch(p -> p.joinType().isOuter());
```

Comment:
Personally, I'm not a big fan of this change, because you need a longer path to trace back here and find why it cannot be pushed down.

Reply:
On the other hand, it's kind of nice not having to understand the details of other components. You just know that you won't be able to push down the order by, for whatever reasons. I think that is more descriptive and a better abstraction than "isOuterJoin".